[Homekit]  NPE Exception thrown continuously

<!-- Provide a general summary of the issue in the *Title* above -->
<!-- If the issue is related to a binding, please include its short name in -->
<!-- square brackets in the title - Example: "[astro] My issue..." -->

<!-- Important: Please contact the openHAB community forum for questions or -->
<!-- for configuration and usage guidance: https://community.openhab.org -->

<!-- Feel free to delete any comment lines in the template (starting with "<!--") -->


## Current Behavior
<!-- If describing a bug, tell us what happens instead of the expected behavior -->
<!-- Include related log information (preferably debug level) and related configs. -->
<!-- Use file attachment for log and config information longer than a few lines -->
<!-- If suggesting a change/improvement, explain the difference from current behavior -->
<!-- For improvements, discuss at community.openhab.org first and include link to topic -->
After enabling the homekit binding, initially it works.  After some amount of time, this could be 10 mins or even an hour, the homekit functionality ceases to work ( Home apps show devices are "not responding" ).   Aditionally i see this message in my logs , hundreds of times:

```
2019-05-26 09:10:20.677 [INFO ] [impl.connections.SubscriptionManager] - Publishing changes for 1928283748
2019-05-26 09:10:20.679 [INFO ] [impl.connections.SubscriptionManager] - Publishing changes for 1928283748
2019-05-26 09:10:20.682 [ERROR] [impl.connections.SubscriptionManager] - Failed to create new event message
java.lang.NullPointerException: null
	at com.beowulfe.hap.impl.connections.SubscriptionManager.publish(SubscriptionManager.java:98) ~[?:?]
	at com.beowulfe.hap.impl.connections.SubscriptionManager.lambda$addSubscription$0(SubscriptionManager.java:37) ~[?:?]
	at org.openhab.io.homekit.internal.HomekitAccessoryUpdater.lambda$1(HomekitAccessoryUpdater.java:55) ~[?:?]
	at org.eclipse.smarthome.core.items.GenericItem$1.run(GenericItem.java:262) [133:org.openhab.core:2.5.0.201905172320]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:?]
	at java.lang.Thread.run(Thread.java:748) [?:?]
```
This has also been reported on the community forums as well. https://community.openhab.org/t/homekit-the-unsetitemregistry-method-has-thrown-an-exception/70942/5?u=digitaldan


## Possible Solution
<!-- Not obligatory, but suggest a fix/reason for the bug, -->
<!-- or ideas how to implement the addition or change -->
I am guessing that the callback here https://github.com/openhab/openhab2-addons/blob/4ff0808f1859038ff88d9b39a9819f8018866ff4/bundles/org.openhab.io.homekit/src/main/java/org/openhab/io/homekit/internal/HomekitAccessoryUpdater.java#L55 is null, and i could submit a patch that checks for that, but my guess is that would not be solving the core issue of why its null. 

## Steps to Reproduce (for Bugs)
<!-- Provide a link to a live example, or an unambiguous set of steps to -->
<!-- reproduce this bug. Include code to reproduce, if relevant -->
1.  Enable devices in homekit
2. Let run for awhile

## Context
<!-- How has this issue affected you? What are you trying to accomplish? -->
<!-- Providing context helps us come up with a solution that is most useful in the real world -->

## Your Environment
<!-- Include as many relevant details about the environment you experienced the bug in -->
* Version used: (e.g., openhab and addon versions)
openHAB nightly build May 19th (2.5.0.201905191816) 

* Environment name and version (e.g. Chrome 39, node.js 5.4, Java 8, ...):
openjdk version "1.8.0_152"
OpenJDK Runtime Environment (Zulu Embedded 8.25.0.76-linux-aarch32hf) (build 1.8.0_152-b76)
OpenJDK Client VM (Zulu Embedded 8.25.0.76-linux-aarch32hf) (build 25.152-b76, mixed mode, Evaluation)
* Operating System and version (desktop or mobile, Windows 10, Raspbian Jessie, ...):
Debian Jessie, asus tinkerboard

--- a/bundles/org.openhab.io.homekit/src/main/java/org/openhab/io/homekit/internal/HomekitAccessoryRegistry.java
+++ b/bundles/org.openhab.io.homekit/src/main/java/org/openhab/io/homekit/internal/HomekitAccessoryRegistry.java
@@ -72,7 +72,11 @@ class HomekitAccessoryRegistry {
     }
 
     public synchronized void unsetBridge() {
-        this.bridge = null;
+        final HomekitRoot oldBridge = bridge;
+        if (oldBridge != null) {
+            createdAccessories.values().forEach(accessory -> oldBridge.removeAccessory(accessory));
+        }
+        bridge = null;
     }
 
     public synchronized void addRootAccessory(String itemName, HomekitAccessory accessory) {

--- a/bundles/org.openhab.io.homekit/src/main/java/org/openhab/io/homekit/internal/HomekitAccessoryUpdater.java
+++ b/bundles/org.openhab.io.homekit/src/main/java/org/openhab/io/homekit/internal/HomekitAccessoryUpdater.java
@@ -45,6 +45,10 @@ public class HomekitAccessoryUpdater {
         if (item == null) {
             return;
         }
+        if (callback == null) {
+            logger.trace("The received subscription contains a null callback, skipping");
+            return;
+        }
         ItemKey itemKey = new ItemKey(item, key);
         subscriptionsByName.compute(itemKey, (k, v) -> {
             if (v != null) {

